docs/migration-guide.md

```diff
@@ -0,0 +1,732 @@
+
+**Before (Unversioned):**
+```bash
+# Old pipeline updated Deployment directly
+kubectl set image deployment/payment-processor worker=payment-processor:v1.6.0
+```
+
+**After (Versioned):**
+```bash
+# New pipeline updates TemporalWorkerDeployment CRD
+kubectl patch temporalworkerdeployment payment-processor --type='merge' -p='{"spec":{"template":{"spec":{"containers":[{"name":"worker","image":"payment-processor:v1.6.0"}]}}}}'
+```
+
+**What happens next:**
+1. Controller detects the image change
+2. Creates a new Kubernetes `Deployment` (e.g., `payment-processor-v1.6.0`)
+3. Registers the new version with Temporal
+4. Gradually routes traffic according to Progressive strategy
+5. Scales down and cleans up old version once new version is fully deployed
+
+### Step 9: Test Your First Versioned Deployment
+
+Deploy a new version to validate the entire flow:
+
+1. **Make a small, safe change** to your worker code
+2. **Build and push** a new container image
+3. **Update the TemporalWorkerDeployment:**
+   ```bash
+   kubectl patch temporalworkerdeployment payment-processor --type='merge' -p='{"spec":{"template":{"spec":{"containers":[{"name":"worker","image":"payment-processor:v1.6.0"}]}}}}'
+   ```
+4. **Monitor the rollout:**
+   ```bash
+   # Watch the deployment progress
+   kubectl get temporalworkerdeployment payment-processor -w
+   
+   # Check that new deployment is created
+   kubectl get deployments -l temporal.io/managed-by=temporal-worker-controller
+   ```
+5. **Verify in Temporal UI** that traffic is gradually shifting to the new version
+
+## Configuration Reference
+
+### Rollout Strategies
+
+See the [Concepts](concepts.md) document for detailed explanations of rollout strategies. Here are the basic configuration patterns:
+
+**Manual Strategy (Default Behavior):**
+```yaml
+cutover:
+  strategy: Manual
+# Requires manual intervention to promote versions
+```
+
+**Immediate Cutover:**
+```yaml
+cutover:
+  strategy: AllAtOnce
+# Immediately routes 100% traffic to new version when healthy
+```
+
+**Progressive Rollout:**
+```yaml
+cutover:
+  strategy: Progressive
+  steps:
+    - rampPercentage: 1
+      pauseDuration: 5m
+    - rampPercentage: 10  
+      pauseDuration: 10m
+    - rampPercentage: 50
+      pauseDuration: 15m
+  gate:
+    workflowType: "HealthCheck"  # Optional validation workflow
+```
+
+### Sunset Configuration
+
+```yaml
+sunset:
+  scaledownDelay: 1h    # Wait 1 hour after draining before scaling to 0
+  deleteDelay: 24h      # Wait 24 hours after draining before deleting
+```
+
+### Resource Management
+
+**CPU and Memory:**
+```yaml
+template:
+  spec:
+    containers:
+    - name: worker
+      resources:
+        requests:
+          memory: "1Gi"
+          cpu: "500m"
+        limits:
+          memory: "2Gi" 
+          cpu: "1"
+```
+
+**Pod Annotations and Labels:**
+```yaml
+template:
+  metadata:
+    labels:
+      team: payments
+      environment: production
+```
```

Comment:
I feel like this is out of scope, people should read k8s docs on pod spec for this